packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -867,6 +873,10 @@ export class GarbageCollector implements IGarbageCollector {
                 }
                 if (newDeletedNodes.length > 0) {
                     // Call container runtime to delete these nodes and add deleted nodes to this.deletedNodes.
+                    const deletedNodes = this.runtime.deleteUnusedNodes(newDeletedNodes);
+                    // TODO: GC:Validation - verify that deletedNodes deepEquals newDeletedNodes
+                    this.removeDeletedNodesFromGCState(deletedNodes);
+                    deletedNodes.forEach(item => this.deletedNodes.add(item))
```

Comment:
removeDeletedNodesFromGCState already adds them to this.deletedNodes. Maybe just call this.sweepNodes?